Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -162,70 +167,203 @@ static s32 MapWiiSockOptNameToNative(u32 optname)
   return optname;
 }
 
-// u32 values are in little endian (i.e. 0x0100007f means 127.0.0.1)
+struct InterfaceRouting
+{
+  u32 index;
+  in_addr destination;
+  in_addr netmask;
+  in_addr gateway;
+};
+
 struct DefaultInterface
 {
-  u32 inet;       // IPv4 address
-  u32 netmask;    // IPv4 subnet mask
-  u32 broadcast;  // IPv4 broadcast address
+  in_addr inet;                                 // IPv4 address
+  in_addr netmask;                              // IPv4 subnet mask
+  in_addr broadcast;                            // IPv4 broadcast address
+  std::vector<InterfaceRouting> routing_table;  // IPv4 routing table
 };
 
-static std::optional<DefaultInterface> GetSystemDefaultInterface()
+static std::vector<InterfaceRouting> GetSystemInterfaceRouting()
 {
+  std::vector<InterfaceRouting> routing_table;
+
 #ifdef _WIN32
-  std::unique_ptr<MIB_IPFORWARDTABLE> forward_table;
   DWORD forward_table_size = 0;
+  std::unique_ptr<MIB_IPFORWARDTABLE> forward_table;
   if (GetIpForwardTable(nullptr, &forward_table_size, FALSE) == ERROR_INSUFFICIENT_BUFFER)
   {
     forward_table =
         std::unique_ptr<MIB_IPFORWARDTABLE>((PMIB_IPFORWARDTABLE) operator new(forward_table_size));
   }
 
-  std::unique_ptr<MIB_IPADDRTABLE> ip_table;
-  DWORD ip_table_size = 0;
-  if (GetIpAddrTable(nullptr, &ip_table_size, FALSE) == ERROR_INSUFFICIENT_BUFFER)
-  {
-    ip_table = std::unique_ptr<MIB_IPADDRTABLE>((PMIB_IPADDRTABLE) operator new(ip_table_size));
-  }
-
-  // find the interface IP used for the default route and use that
-  NET_IFINDEX ifIndex = NET_IFINDEX_UNSPECIFIED;
   DWORD result = GetIpForwardTable(forward_table.get(), &forward_table_size, FALSE);
   // can return ERROR_MORE_DATA on XP even after the first call
   while (result == NO_ERROR || result == ERROR_MORE_DATA)
   {
     for (DWORD i = 0; i < forward_table->dwNumEntries; ++i)
     {
```

Comment:
Can't a `std::span` and range-based for loop be used here instead of iterating over the table manually?

```C++
const std::span<MIB_IPFORWARDROW> table(forward_table->table, forward_table->dwNumEntries);
for (const auto& entry : table)
{
  routing_table.emplace_back(...);
}
```